homeassistant/components/music_assistant/media_player.py

```diff
@@ -276,22 +276,36 @@ async def async_on_update(self) -> None:
             self._attr_state = MediaPlayerState(player.state.value)
         else:
             self._attr_state = MediaPlayerState(STATE_OFF)
+
         group_members_entity_ids: list[str] = []
+        coordinator_id: str | None = None
+        group_childs: set[str] = set()
         if player.group_childs:
+            # This player is the coordinator, with children players
+            coordinator_id = self.player_id
+            group_childs = set(player.group_childs) - {self.player_id}
+        elif player.synced_to:
+            # This player is a child player, with a coordinator
+            coordinator_id = player.synced_to
+            # Do not list siblings. It's unclear if we should list all siblings or just this player
+            group_childs = {self.player_id}
+
+        if coordinator_id and group_childs:
+            # Set the coordinator as the first entity by convention
+            group_members = [coordinator_id, *sorted(group_childs)]
             # translate MA group_childs to HA group_members as entity id's
             entity_registry = er.async_get(self.hass)
             group_members_entity_ids = [
                 entity_id
-                for child_id in player.group_childs
+                for child_id in group_members
                 if (
                     entity_id := entity_registry.async_get_entity_id(
                         self.platform.domain, DOMAIN, child_id
                     )
                 )
             ]
-        # NOTE: we sort the group_members for now,
-        # until the MA API returns them sorted (group_childs is now a set)
-        self._attr_group_members = sorted(group_members_entity_ids)
```

Comment:
this comment is old news, the group members are sorted since a while now (no longer a set)

Reply:
Thank you for the information!

This stale comment was the main reason for all the set manipulation and then re-sorting in my PR. Since I needed to guarantee that the coordinator was first in the list, and as you said in a different comment, sets are unordered, so I needed to remove the coordinator from the set, then order them semi-manually.